lib/ansible/modules/cloud/cloudstack/cs_storage_pool.py

```diff
@@ -0,0 +1,399 @@
+    returned: success
+    type: string
+    sample: Up
+tags:
+    description: the Tags for the storage pool
+    returned: success
+    type: string
+    sample: rbd
+'''
+
+# import cloudstack common
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    CloudStackException,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+
+class AnsibleCloudStackStoragePool(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackStoragePool, self).__init__(module)
+        self.returns = {
+            'id': 'id',
+            'capacityiops': 'capacity_iops',
+            'zonename': 'zone_name',
+            'zoneid': 'zone_id',
+            'podname': 'pod_name',
+            'podid': 'pod_id',
+            'clustername': 'cluster_name',
+            'clusterid': 'cluster_id',
+            'disksizeallocated': 'disk_size_allocated',
+            'disksizetotal': 'disk_size_total',
+            'disksizeused': 'disk_size_used',
+            'scope': 'scope',
+            'hypervisor': 'hypervisor',
+            'state': 'state',
+            'tags': 'tags',
+        }
+        self.storage_pool = None
+        self.zone = None
+        self.pod = None
+        self.cluster = None
+
+    def _get_common_args(self):
+        args = {
+            'name': self.module.params.get('name'),
+            'url': self.module.params.get('storage_url'),
+            'zoneid': self.get_zone(key='id'),
+            'provider': self.module.params.get('provider'),
+            'scope': self.module.params.get('scope'),
+            'hypervisor': self.module.params.get('hypervisor'),
+            'capacitybytes': self.module.params.get('capacity_bytes'),
+            'capacityiops': self.module.params.get('capacity_iops'),
+        }
+        return args
+
+    def get_storage_pool(self, key=None):
+        if self.storage_pool is None:
+            zoneid = self.get_zone(key='id')
+            clusterid = self.get_cluster(key='id')
+            podid = self.get_pod(key='id')
+
+            args = {
+                'zoneid': zoneid,
+                'podid': podid,
+                'clusterid': clusterid,
+                'name': self.module.params.get('name'),
+            }
+
+            res = self.cs.listStoragePools(**args)
+            if 'errortext' in res:
+                self.module.fail_json(msg="Failed: '%s'" % res['errortext'])
+
+            elif 'storagepool' not in res:
+                return None
+
+            self.storage_pool = res['storagepool'][0]
+
+        return self.storage_pool
+
+    def present_storage_pool(self):
+        pool = self.get_storage_pool()
+        if pool:
+            pool = self._update_storage_pool()
+        else:
+            pool = self._create_storage_pool()
+        return pool
+
+    def _create_storage_pool(self):
+
+        cluster = self.module.params.get('cluster')
```

Comment:
can be simplified to 
~~~
cluster = self.get_cluster(key='id')
~~~
and following lines can be removed
~~~
if cluster:
    args['clusterid'] = self.get_cluster(key='id')
~~~